### PR TITLE
Update of PERCENT keyword in the LIMIT clause(#2671)

### LINKED ISSUE #2582: Addition of PERCENT keyword in the LIMIT clause
Both Oracle and MsSQL have the ability to limit rows by either the count, or with the `PERCENT` keyword, by the percent of rows returned. 

The syntax for the [`LIMIT`](https://duckdb.org/docs/sql/query_syntax/limit) clause would change to allowing the optional `PERCENT` keyword directly after the count/expr:

LIMIT expr [{PERCENT|%}] OFFSET expr

@Mytherin -- is this something that I could work to implement, or something you want to review with the team?

### PR DESCRIPTION
**Pull request by chenxi8611**
I've closed previous PR and made new PR. I've removed unnecessary file (src.zip) and commit history.
Please review it.

### CODE REVIEW

File: src/execution/operator/helper/physical_limit_percent.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		if (limit == 0) {
+			return;
+		}
+	}
+
+	if (current_offset >= limit) {
+		return;
+	}
+
+	while (state.chunk_idx < gstate.data.ChunkCount()) {
```
Comment (Mytherin) [inline src/execution/operator/helper/physical_limit_percent.cpp:123]: This loop should no longer be necessary, no?

**Review by Mytherin — COMMENTED** [review]
Thanks! Looks good. One more minor comment, then this is ready to merge:

File: src/execution/operator/helper/physical_limit_percent.cpp (hunk 1 above)
Comment (chenxi8611) [reply src/execution/operator/helper/physical_limit_percent.cpp:123]: Good, you are right. I will remove this loop.

**Comment by chenxi8611** [comment]
I've removed the unnecessary loop, please review it again.

**Review by Mytherin — APPROVED** [review]
Everything looks good to me now. Will merge after the tests pass.

**Pull request merged**